core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -663,7 +673,7 @@ class SnappyParser(session: SnappySession)
     SELECT ~ (DISTINCT ~> trueFn).? ~
     (namedExpression + commaSep) ~
     (FROM ~ ws ~ relations).? ~
-    (WHERE ~ expression).? ~
+    (WHERE ~ (test(setTokenize) ~ expression ~ test(setNoTokenize))).? ~
```

Comment:
Yes this is clean. I was not happy with this way either but didn't know about MATCH special rule.